### ISSUE TITLE
unreasonable ConstraintViolationError when using torch dynamo to compile torch model

### ISSUE THREAD

**Issue by Jason3900** [pytorch/pytorch#143463]
### 🐛 Describe the bug

I'm using torch dynamo backend to compile model to export to tensorrt.
```python
    inputs = [torch.randn(1, 3, 28, 288, 512).cuda().to(torch.float16)]
    
    dynamic_h = torch.export.Dim("dim_3", min=224, max=640)
    dynamic_w = torch.export.Dim("dim_4", min=224, max=640)
    dynamic_t = torch.export.Dim("dim_2", min=1, max=200)
    dynamic_shapes={"x": {2:dynamic_t, 3: dynamic_h, 4: dynamic_w}}

    exp_program = torch.export.export(encoder_model, args=tuple(inputs), dynamic_shapes=dynamic_shapes, strict=True)

    trt_model = torch_tensorrt.dynamo.compile(
        exported_program=exp_program,
        assume_dynamic_shape_support=True,
        inputs=inputs, 
        make_refitable=True, 
        disable_tf32=True, 
        debug=True, 
        enabled_precisions={torch.half, torch.float},
        torch_executed_ops={},
        min_block_size=17,
        truncate_double=True,
        use_python_runtime=False)
```
I've confirmed that the h, w dim can be dynamic without constraint error but the dim_2 cannot. It's quite strange.
I trace the cause and found it's in here
```
class GroupNormSpatial(nn.Module):
    """GroupNorm with spatial dimensions ignored."""
    def __init__(self, num_groups, num_channels, epsilon: float = 1e-5, affine=True):
        super().__init__()
        # affine=False # TODO: for tensorrt only
        self.norm_fn = nn.GroupNorm(num_groups=num_groups, num_channels=num_channels, eps=epsilon, affine=affine)

    def forward(self, inputs: torch.Tensor) -> torch.Tensor:
        if int(inputs.ndim) == 5: # video
            b, c, t, h, w = inputs.shape
            inputs = inputs.permute(0,2,1,3,4).flatten(start_dim=0, end_dim=1) # ERROR
            out = self.norm_fn(inputs)
            out = out.reshape(b, t, c, h, w).permute(0,2,1,3,4) # ERROR
            return out
        else: # Image, b c h w -> b c h w
            out = self.norm_fn(inputs)
            return out
```

```
I1218 02:40:31.595000 155463 torch/_utils_internal.py:116] [0/0] CompilationMetrics(compile_id='0/0', frame_key='1', co_name='forward', co_filename='xxx,py', co_firstlineno=50, cache_size=0, accumulated_cache_size=0, guard_count=None, shape_env_guard_count=None, graph_op_count=None, graph_node_count=None, graph_input_count=None, start_time=1734489622.2484767, entire_frame_compile_time_s=None, backend_compile_time_s=None, inductor_compile_time_s=None, code_gen_time_s=None, fail_type="<class 'torch.fx.experimental.symbolic_shapes.ConstraintViolationError'>", fail_reason='Constraints violated (dim_2)! For more information, run with TORCH_LOGS="+dynamic".\n  - Not all values of dim_2 = L[\'x\'].size()[2] in the specified range dim_2 <= 200 satisfy the generated guard Ne(Mod(1, ((L[\'x\'].size()[2] - 1)//2) + 1), 0).\n  - Not all values of dim_2 = L[\'x\'].size()[2] in the specified range dim_2 <= 200 satisfy the generated guard Ne(Mod(1, ((L[\'x\'].size()[2] - 1)//4) + 1), 0).\n  - Not all values of dim_2 = L[\'x\'].size()[2] in the specified range dim_2 <= 200 satisfy the generated guard 9 <= L[\'x\'].size()[2] and L[\'x\'].size()[2] <= 200', fail_user_frame_filename=None, fail_user_frame_lineno=None, non_compliant_ops=set(), compliant_custom_ops=set(), restart_reasons=set(), dynamo_time_before_restart_s=9.347490787506104, has_guarded_code=False, possibly_missed_reinplacing_opportunities=None)
```

### Versions

I'm using ngc torch `nvcr.io/nvidia/pytorch:24.10-py3`

**Comment by ezyang** [comment]
So, uh, what are your TORCH_LOGS=dynamic?

**Comment by Jason3900** [comment]
uh, it prints out lot of messages, but the main problem is about constraints introduced by groupnorm, others are just info messages.

**Comment by ydwu4** [comment]
Can the model export? Or the program fails at torch_tensorrt.dynamo.compile step?

Not sure how the whole model looks like. Reproing with the GroupNormSpatial model:
```python
import torch
import torch.nn as nn
# Define the GroupNormSpatial class (same as your code)
class GroupNormSpatial(nn.Module):
    """GroupNorm with spatial dimensions ignored."""
    def __init__(self, num_groups, num_channels, epsilon: float = 1e-5, affine=True):
        super().__init__()
        # affine=False # TODO: for tensorrt only
        self.norm_fn = nn.GroupNorm(num_groups=num_groups, num_channels=num_channels, eps=epsilon, affine=affine)
    def forward(self, inputs: torch.Tensor) -> torch.Tensor:
        if int(inputs.ndim) == 5: # video
            b, c, t, h, w = inputs.shape
            inputs = inputs.permute(0,2,1,3,4).flatten(start_dim=0, end_dim=1) 
            out = self.norm_fn(inputs)
            out = out.reshape(b, t, c, h, w).permute(0,2,1,3,4) 
            return out
        else: # Image, b c h w -> b c h w
            out = self.norm_fn(inputs)
            return out

# Create an instance of GroupNormSpatial
num_groups = 16
num_channels = 64
gn = GroupNormSpatial(num_groups, num_channels)
# Create a random input tensor for video data (5D)
video_input = torch.randn(1, 64, 10, 224, 224)  # batch_size, channels, time, height, width
dynamic_w = torch.export.Dim("dim_4", min=224, max=640)
dynamic_h = torch.export.Dim("dim_3", min=224, max=640)
dynamic_t = torch.export.Dim("dim_2", min=1, max=200)
ep = torch.export.export(gn, (video_input,), dynamic_shapes={
    "inputs": {2:dynamic_t, 3: dynamic_h, 4: dynamic_w}
})
print(ep)
```
The TORCH_LOGS=+dynamic log:
```
I1219 11:56:14.542000 3134762 torch/fx/experimental/symbolic_shapes.py:3194] [0/0] create_env
V1219 11:56:14.572000 3134762 torch/fx/experimental/symbolic_shapes.py:5804] [0/0] _update_var_to_range s0 = VR[2, 200] (update)
I1219 11:56:14.573000 3134762 torch/fx/experimental/symbolic_shapes.py:4425] [0/0] create_symbol s0 = 10 for L['inputs'].size()[2] [2, 200] (_dynamo/variables/builder.py:2872 in <lambda>), for more info run with TORCHDYNAMO_EXTENDED_DEBUG_CREATE_SYMBOL="s0" or to suppress this message run with TORCHDYNAMO_EXTENDED_ADVICE="0"
V1219 11:56:14.574000 3134762 torch/fx/experimental/symbolic_shapes.py:5804] [0/0] _update_var_to_range s1 = VR[224, 640] (update)
I1219 11:56:14.575000 3134762 torch/fx/experimental/symbolic_shapes.py:4425] [0/0] create_symbol s1 = 224 for L['inputs'].size()[3] [224, 640] (_dynamo/variables/builder.py:2872 in <lambda>), for more info run with TORCHDYNAMO_EXTENDED_DEBUG_CREATE_SYMBOL="s1" or to suppress this message run with TORCHDYNAMO_EXTENDED_ADVICE="0"
V1219 11:56:14.576000 3134762 torch/fx/experimental/symbolic_shapes.py:5804] [0/0] _update_var_to_range s2 = VR[224, 640] (update)
I1219 11:56:14.576000 3134762 torch/fx/experimental/symbolic_shapes.py:4425] [0/0] create_symbol s2 = 224 for L['inputs'].size()[4] [224, 640] (_dynamo/variables/builder.py:2872 in <lambda>), for more info run with TORCHDYNAMO_EXTENDED_DEBUG_CREATE_SYMBOL="s2" or to suppress this message run with TORCHDYNAMO_EXTENDED_ADVICE="0"
V1219 11:56:14.577000 3134762 torch/fx/experimental/symbolic_shapes.py:6616] [0/0] runtime_assert True == True [statically known]
V1219 11:56:14.651000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(s2, 1) == True [statically known]
V1219 11:56:14.652000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(s1, 1) == True [statically known]
V1219 11:56:14.662000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Eq(s0*s1*s2, s1*s2) == False [statically known]
V1219 11:56:14.665000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(s0*s1*s2, s1*s2) == True [statically known]
V1219 11:56:14.667000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval 1 < s1*s2 == True [statically known]
V1219 11:56:14.670000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval s2 < s1*s2 == True [statically known]
V1219 11:56:14.671000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(s0, 1) == True [statically known]
V1219 11:56:14.678000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(s1*s2, 64*s0*s1*s2) == True [statically known]
V1219 11:56:14.678000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Eq(s0, 1) == False [statically known]
V1219 11:56:14.679000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval 1 < s0 == True [statically known]
V1219 11:56:14.680000 3134762 torch/fx/experimental/symbolic_shapes.py:6616] [0/0] runtime_assert True == True [statically known]
V1219 11:56:14.682000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(Mod(1, s0), 0) == True [statically known]
V1219 11:56:14.692000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval s1*s2 > 1 == True [statically known]
V1219 11:56:14.700000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Eq(s1*s2, s1) == False [statically known]
V1219 11:56:14.705000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(Mod(s1, s1*s2), 0) == True [statically known]
V1219 11:56:14.705000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Eq(s1, 1) == False [statically known]
V1219 11:56:14.707000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval s2 > 1 == True [statically known]
V1219 11:56:14.718000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Eq(s1*s2, 1) == False [statically known]
V1219 11:56:14.720000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(s1*s2, 1) == True [statically known]
V1219 11:56:14.727000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Eq(s1, s1*s2) == False [statically known]
V1219 11:56:14.728000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Ne(s1*s2, s1) == True [statically known]
V1219 11:56:14.731000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] [0/0] eval Eq(s2, 1) == False [statically known]
I1219 11:56:14.754000 3134762 torch/fx/experimental/symbolic_shapes.py:4549] [0/0] produce_guards
V1219 11:56:14.754000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].size()[0] 1 None
V1219 11:56:14.754000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].size()[1] 64 None
V1219 11:56:14.755000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].size()[2] s0 StrictMinMaxConstraint(warn_only=False, vr=VR[1, 200])
V1219 11:56:14.755000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].size()[3] s1 StrictMinMaxConstraint(warn_only=False, vr=VR[224, 640])
V1219 11:56:14.755000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].size()[4] s2 StrictMinMaxConstraint(warn_only=False, vr=VR[224, 640])
V1219 11:56:14.755000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].stride()[0] 64*s0*s1*s2 None
V1219 11:56:14.755000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].stride()[1] s0*s1*s2 None
V1219 11:56:14.756000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].stride()[2] s1*s2 None
V1219 11:56:14.756000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].stride()[3] s2 None
V1219 11:56:14.756000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].stride()[4] 1 None
V1219 11:56:14.756000 3134762 torch/fx/experimental/symbolic_shapes.py:4757] [0/0] track_symint L['inputs'].storage_offset() 0 None
V1219 11:56:14.821000 3134762 torch/fx/experimental/symbolic_shapes.py:6414] eval Ne(s0*s1*s2, 1) == True [statically known]
ExportedProgram:
    class GraphModule(torch.nn.Module):
        def forward(self, p_norm_fn_weight: "f32[64]", p_norm_fn_bias: "f32[64]", inputs: "f32[1, 64, s0, s1, s2]"):
             # 
            sym_size_int_3: "Sym(s0)" = torch.ops.aten.sym_size.int(inputs, 2)
            sym_size_int_4: "Sym(s1)" = torch.ops.aten.sym_size.int(inputs, 3)
            sym_size_int_5: "Sym(s2)" = torch.ops.aten.sym_size.int(inputs, 4)
            
             # File: /data/users/yidi/pytorch/test.py:13 in forward, code: inputs = inputs.permute(0,2,1,3,4).flatten(start_dim=0, end_dim=1)
            permute: "f32[1, s0, 64, s1, s2]" = torch.ops.aten.permute.default(inputs, [0, 2, 1, 3, 4]);  inputs = None
            flatten: "f32[s0, 64, s1, s2]" = torch.ops.aten.flatten.using_ints(permute, 0, 1);  permute = None
            
             # File: /data/users/yidi/pytorch/test.py:14 in forward, code: out = self.norm_fn(inputs)
            group_norm: "f32[s0, 64, s1, s2]" = torch.ops.aten.group_norm.default(flatten, 16, p_norm_fn_weight, p_norm_fn_bias);  flatten = p_norm_fn_weight = p_norm_fn_bias = None
            
             # File: /data/users/yidi/pytorch/test.py:15 in forward, code: out = out.reshape(b, t, c, h, w).permute(0,2,1,3,4)
            reshape: "f32[1, s0, 64, s1, s2]" = torch.ops.aten.reshape.default(group_norm, [1, sym_size_int_3, 64, sym_size_int_4, sym_size_int_5]);  group_norm = sym_size_int_3 = sym_size_int_4 = sym_size_int_5 = None
            permute_1: "f32[1, 64, s0, s1, s2]" = torch.ops.aten.permute.default(reshape, [0, 2, 1, 3, 4]);  reshape = None
            return (permute_1,)
            
Graph signature: ExportGraphSignature(input_specs=[InputSpec(kind=<InputKind.PARAMETER: 2>, arg=TensorArgument(name='p_norm_fn_weight'), target='norm_fn.weight', persistent=None), InputSpec(kind=<InputKind.PARAMETER: 2>, arg=TensorArgument(name='p_norm_fn_bias'), target='norm_fn.bias', persistent=None), InputSpec(kind=<InputKind.USER_INPUT: 1>, arg=TensorArgument(name='inputs'), target=None, persistent=None)], output_specs=[OutputSpec(kind=<OutputKind.USER_OUTPUT: 1>, arg=TensorArgument(name='permute_1'), target=None)])
Range constraints: {s0: VR[1, 200], s1: VR[224, 640], s2: VR[224, 640]}

```

which looks fine i think? It would be helpful if we could have all the TORCH_LOGS=+dynamic logs.
